omod/src/test/java/org/openmrs/module/registrationapp/fragment/controller/RegistrationSummaryFragmentControllerTest.java

```diff
@@ -0,0 +1,140 @@
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Patient;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.module.appframework.context.AppContextModel;
+import org.openmrs.module.appframework.domain.AppDescriptor;
+import org.openmrs.module.appframework.domain.Extension;
+import org.openmrs.module.appframework.service.AppFrameworkService;
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.module.emrapi.patient.PatientDomainWrapper;
+import org.openmrs.module.registrationapp.RegistrationAppConstants;
+import org.openmrs.module.registrationapp.fragment.controller.summary.RegistrationSummaryFragmentController;
+import org.openmrs.ui.framework.fragment.FragmentConfiguration;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+
+public class RegistrationSummaryFragmentControllerTest {
+
+    private RegistrationSummaryFragmentController regSummaryFragmentcontroller;
+
+    private AppDescriptor appDescriptor;
+    
+    private FragmentConfiguration fragmentconfig;
+
+    private Patient patient;
+    
+    private AdministrationService administrationService;
+    
+    private UiSessionContext sessionContext;
+    
+    private PatientDomainWrapper patientDomainWrapper;
+    
+    private AppFrameworkService appFrameworkService;
+    
+    private List<Extension> extnList;
+    
+    @Before
+    public void setUp() throws Exception {
+    	patientDomainWrapper = new PatientDomainWrapper(); 
+    	patient = new Patient();
+    	extnList = new ArrayList<Extension>();
+    	
+    	fragmentconfig = new FragmentConfiguration();
+    	fragmentconfig.put("patient", patient);
+    	fragmentconfig.put("appId", "referenceapplication.registrationapp.registerPatient");
+
+        regSummaryFragmentcontroller = new RegistrationSummaryFragmentController();
+        
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream("registration_app.json");    	
+    	List<AppDescriptor> appDescriptors = new ObjectMapper().readValue(inputStream, new TypeReference<List<AppDescriptor>>() {});
+    	appDescriptor = appDescriptors.get(0);
+    	
+        appFrameworkService = mock(AppFrameworkService.class);
+        when(appFrameworkService.getExtensionsForCurrentUser(anyString(), any(AppContextModel.class))).thenReturn(extnList);
+        when(appFrameworkService.getApp(eq((String) fragmentconfig.get("appId")))).thenReturn(appDescriptor);
+        
+        administrationService = mock(AdministrationService.class);
+
+        sessionContext = mock(UiSessionContext.class);
+        when(sessionContext.generateAppContextModel()).thenReturn(new AppContextModel());
+    }
+
+    @Test
+    public void controller_shouldAddAutoGeneratedDistributedSummaryWidgets() throws Exception {
```

Comment:
→ `controller_shouldDistributedAutoGeneratedSummaryWidgets`

Reply:
Thank you @mks-d, method name improved.